### PR TITLE
Improve completions after configurable keyword in ModulePart cxt [2201.2.x]

### LINKED ISSUE #36323: Type suggestions are not provided after `configurable` keyword when its followed by a function
Consider the following scenario. 
```
configurable <cursor>

public function main() {
    
}
``` 
Here the completion items include module level completions items (like main function, type defs, service declarations, etc). This needs to be fixed to suggest types instead.

_Originally posted by @mindula in https://github.com/ballerina-platform/ballerina-lang/issues/36189#issuecomment-1131190584_

### PR DESCRIPTION
**Pull request by LakshanWeerasinghe**
## Purpose
$subject 

Fixes #36323
Related PR #37099, #38214(master)

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [x] Added necessary tests
  - [x] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [x] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: language-server/modules/langserver-core/src/test/resources/completion/module_var_context/source/source17.bal [hunk 1]
```diff
@@ -0,0 +1,7 @@
+import ballerina/module1;
+
+configurable module1:
```
Comment (IMS94) [inline language-server/modules/langserver-core/src/test/resources/completion/module_var_context/source/source17.bal:3]: Shouldn't we sort the members of `module1` based on subtypes of `anydata`?

**Review by IMS94 — CHANGES_REQUESTED** [review]
Changes looks good. Added few minor comments

File: language-server/modules/langserver-core/src/test/resources/completion/xml_typedesc_context/config/config3.json [hunk 2]
```diff
@@ -9,31 +9,43 @@
...
     },
     {
       "label": "ProcessingInstruction",
       "kind": "TypeParameter",
       "detail": "Processinginstruction",
-      "sortText": "BN",
+      "documentation": {
+        "left": "Type for singleton processing instructions.\nBuilt-in subtype of xml."
+      },
+      "sortText": "G",
```
[the remark this answers is not in the document]
Comment (LakshanWeerasinghe) [reply language-server/modules/langserver-core/src/test/resources/completion/xml_typedesc_context/config/config3.json:26]: These completion items belong to the `lang` module. I have added a check in `SortingUtil`.

[the remark this answers is not in the document]
Comment (LakshanWeerasinghe) [reply language-server/modules/langserver-core/src/test/resources/completion/xml_typedesc_context/config/config3.json:26]: In this implementation we have added the module symbol to the completion items coming for `lang` module.

[the remark this answers is not in the document]
Comment (IMS94) [reply language-server/modules/langserver-core/src/test/resources/completion/xml_typedesc_context/config/config3.json:26]: I see. No problem then.

File: language-server/modules/langserver-core/src/test/resources/completion/module_var_context/source/source17.bal (hunk 1 above)
Comment (LakshanWeerasinghe) [reply language-server/modules/langserver-core/src/test/resources/completion/module_var_context/source/source17.bal:3]: Yes. I have updated the sorting logic now. And added two test cases for both `ModulePartNodeContext` and `ModuleVarNodeContext`.

Comment (IMS94) [reply language-server/modules/langserver-core/src/test/resources/completion/module_var_context/source/source17.bal:3]: Great! :+1:

**Review by IMS94 — APPROVED** [review]

File: language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/common/utils/CommonUtil.java [hunk 3]
```diff
@@ -622,4 +621,19 @@ public static io.ballerina.tools.text.TextEdit getTextEdit(SyntaxTree syntaxTree
         int end = textDocument.textPositionFrom(LinePosition.from(endPos.getLine(), endPos.getCharacter()));
         return io.ballerina.tools.text.TextEdit.from(TextRange.from(start, end - start), textEdit.getNewText());
     }
+
+    /**
+     * Get the last appearing qualifier token of a give node.
```
Comment (mindula) [inline language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/common/utils/CommonUtil.java:626]:
```suggestion
     * Get the last appearing qualifier token of a given node.
```

File: language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/completions/providers/context/ModulePartNodeContext.java [hunk 4]
```diff
@@ -164,9 +174,29 @@ private List<LSCompletionItem> getModulePartContextItems(BallerinaCompletionCont
                 .getServiceTemplates(context));
         return completionItems;
     }
-
+
```
Comment (mindula) [inline language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/completions/providers/context/ModulePartNodeContext.java:177]: Let's remove these additional spaces
```suggestion

```

File: language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/completions/util/SortingUtil.java [hunk 5]
```diff
@@ -110,6 +113,22 @@ public static boolean isTypeCompletionItem(LSCompletionItem item) {
                 && ((StaticCompletionItem) item).kind() == StaticCompletionItem.Kind.TYPE);
     }
 
+    /**
+     * Check whether the item is an associated lang lib module completion item.
```
Comment (mindula) [inline language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/completions/util/SortingUtil.java:117]:
```suggestion
     * Check if the item is an associated lang lib module completion item.
```

File: language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/completions/util/SortingUtil.java [hunk 6]
```diff
@@ -199,6 +218,12 @@ public static String genSortTextForTypeDescContext(BallerinaCompletionContext co
             String orgName = symbolModule.isPresent() ? symbolModule.get().id().orgName() : "";
             String moduleName = symbolModule.isPresent() ? symbolModule.get().id().moduleName() : "";
 
+            // Case 7
+            // Skip Types coming from lang.value (StrandData, Thread)
```
Comment (mindula) [inline language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/completions/util/SortingUtil.java:222]:
```suggestion
            // Skip types coming from lang.value (StrandData, Thread)
```

**Review by IMS94 — DISMISSED** [review]

**Comment by IMS94** [comment]
@LakshanWeerasinghe `ExpressionContextTest > test[126](module_ctx_config5.json, expression_context) FAILED` there's a test failure

**Comment by IMS94** [comment]
Closing this since this won't be added to 2201.2.x. Added to master in #38214

**Pull request closed without merging**